src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -170,6 +170,11 @@ private bool CheckGlobalConditions(Log log, State state)
                 return log.Fail("CopyAlwaysItemExists", "Item '{0}' has CopyToOutputDirectory set to 'Always', not up to date.", _configuredProject.UnconfiguredProject.MakeRooted(copyAlwaysItemPath));
             }
 
+
+            if (state.AdditionalDependentFilesAddedOrRemoved)
```

Comment:
The comparison should be against the last output time. See the existing check against `LastItemsChangedAtUtc` in `CheckInputsAndOutputs`.